src/redux/books/books.js

```diff
@@ -13,9 +34,7 @@ export default function reducer(state = books, action = {}) {
       ];
 
     case DELETE:
-      return [
-        books.map((book) => book.id !== action.id),
-      ];
+      return [...books.filter((book) => (book.id !== action.id))];
```

Comment:
- [ ] When we try to delete books in the current state of the app, we are not able to delete all books. The bug is caused by the return statement in the DELETE action call.

  Every time we call our actions, we need to reference the state itself, not the initial state(books in this case). So the return statement should be like this 👇🏼:
  
  ```JavaScript
  case DELETE:
      return [...state.filter((book) => book.id !== action.id)];
  ```
  
  Please consider refactoring the call to the `ADDNEW `action too. Instead of adding to books, add data to the apps state. Do something like this 👇🏼:
  
  ```JavaScript
  case ADDNEW:
      return [...state, action.payload];
  ```

Reply:
Thanks @Lameck1, I will refactor them.